src/cmd/run.go

```diff
@@ -58,20 +57,17 @@ func doRun(cmd *cobra.Command, args []string) {
 
 		pkg.StartMetricsServer(string(runner.Id), viper.GetInt("metrics-port"))
 
-		if viper.GetBool("scaling-enabled") {
-			config, err := pkg.GetKubernetesConfig()
-			pkg.CheckErr(err)
-
-			k8sClient := clientset.NewForConfigOrDie(config)
+		ctx := signal.Init(context.Background())
 
-			log.Info().Msgf("electing leader...")
-			go electLeader(k8sClient, runner.Id)
+		if viper.GetBool("scaling-enabled") {
+			leaseLockName := viper.GetString("runner-deployment")
```

Comment:
I removed the `electLeader` function since it was just doing viper calls.

The go routine was moved down into the function and now we properly exit if leader election setup fails for some reason.